shell/common/shell.cc

```diff
@@ -734,34 +733,22 @@ void Shell::OnPlatformViewDestroyed() {
   bool should_post_raster_task = task_runners_.GetRasterTaskRunner() !=
                                  task_runners_.GetPlatformTaskRunner();
 
-  auto ui_task = [engine = engine_->GetWeakPtr(),
-                  raster_task_runner = task_runners_.GetRasterTaskRunner(),
-                  raster_task, should_post_raster_task, &latch]() {
-    if (engine) {
-      engine->OnOutputSurfaceDestroyed();
-    }
+  // Step 0: tell the engine that its output
+  // surface is about to go away.
+  have_surface_.store(false);
```

Comment:
are there any other components that need to keep this parameter in sync? What is the lifecycle of a surface now? I'm wondering if we need this given `have_surface_` and `StopAnimator` are all invoked by engine.cc `OnOutputSurfaceDestroyed`. Can you explain how these fields are different from those? It might make sense to unify them.

Reply:
have_surface_ is used in engine and pause is used in animator. I think one way we can unify them is to expose a public method in animator like `IsRunning`. So the engine can use `animator_.IsRunning` to determine its behaviors. WDYT?